app/templates/entry/server-entry.js

```diff
@@ -137,16 +137,23 @@ export default ssrContext => {
         redirectBrowser(url, router, reject, httpStatusCode)
       }
 
-      App.preFetch !== void 0 && matchedComponents.unshift(App)
-
+      // Class components return the component options (and the preFetch hook) inside __c property
+      if (typeof App.preFetch === 'function' || App.__c && typeof App.__c.preFetch === 'function') {
+        matchedComponents.unshift(App)
+      }
       // Call preFetch hooks on components matched by the route.
       // A preFetch hook dispatches a store action and returns a Promise,
       // which is resolved when the action is complete and store state has been
       // updated.
       matchedComponents
-      .filter(c => c && c.preFetch)
+      .filter(c => {
```

Comment:
Why not simplify this to be 
```ts
.filter(c => {
        // Class components return the component options (and the preFetch hook) inside __c property
        return (typeof c.preFetch === 'function') || (c.__c && typeof c.__c.preFetch === 'function')         
       #})
```

Reply:
I fond my version a bit more readable, less thinking about what it does. But if you insist I can change it.

Reply:
I'd use the short version to keep the codebase style coherent, removing braces too.
Even if I agree that splitting it into multiple lines is more readable

Reply:
I used the short versions of the filter now.